src/main/java/appeng/util/item/FluidList.java

```diff
@@ -0,0 +1,219 @@
+		final IAEFluidStack opt = option.copy();
+
+		this.putFluidRecord( opt );
+	}
+
+	@Override
+	public IAEFluidStack findPrecise( IAEFluidStack i )
+	{
+		if( this.isStackNull( i ) )
+		{
+			return null;
+		}
+
+		final IAEFluidStack is = this.getFluidRecord( i );
+
+		return is;
+	}
+
+	@Override
+	public Collection<IAEFluidStack> findFuzzy( IAEFluidStack filter, FuzzyMode fuzzy )
+	{
+		if( this.isStackNull( filter ) )
+		{
+			return Collections.emptyList();
+		}
+
+		final List<IAEFluidStack> result = Lists.newArrayList();
+
+		if( filter.equals( this ) )
+		{
+			result.add( filter );
+		}
+
+		return result;
+
+	}
+
+	@Override
+	public boolean isEmpty()
+	{
+		return !this.iterator().hasNext();
+	}
+
+	@Override
+	public void addStorage( IAEFluidStack option )
+	{
+		if( this.isStackNull( option ) )
+		{
+			return;
+		}
+
+		final IAEFluidStack st = this.getFluidRecord( option );
+
+		if( st != null )
+		{
+			st.incStackSize( option.getStackSize() );
+			return;
+		}
+
+		final IAEFluidStack opt = option.copy();
+		this.putFluidRecord( opt );
+	}
+
+	/*
+	 * public synchronized void clean() { Iterator<StackType> i = iterator(); while (i.hasNext()) { StackType AEI =
+	 * i.next(); if ( !AEI.isMeaningful() ) i.remove(); } }
+	 */
+
+	@Override
+	public void addCrafting( IAEFluidStack option )
+	{
+		if( this.isStackNull( option ) )
+		{
+			return;
+		}
+
+		final IAEFluidStack st = this.getFluidRecord( option );
+
+		if( st != null )
+		{
+			st.setCraftable( true );
+			return;
+		}
+
+		final IAEFluidStack opt = option.copy();
+		opt.setStackSize( 0 );
+		opt.setCraftable( true );
+
+		this.putFluidRecord( opt );
+	}
+
+	@Override
+	public void addRequestable( IAEFluidStack option )
+	{
+		if( this.isStackNull( option ) )
+		{
+			return;
+		}
+
+		final IAEFluidStack st = this.getFluidRecord( option );
+
+		if( st != null )
+		{
+			st.setCountRequestable( st.getCountRequestable() + option.getCountRequestable() );
+			return;
+		}
+
+		final IAEFluidStack opt = option.copy();
+		opt.setStackSize( 0 );
+		opt.setCraftable( false );
+		opt.setCountRequestable( opt.getCountRequestable() );
+
+		this.putFluidRecord( opt );
+	}
+
+	@Override
+	public IAEFluidStack getFirstItem()
+	{
+		for( IAEFluidStack stackType : this )
+		{
+			return stackType;
+		}
+		return null;
+	}
+
+	@Override
+	public int size()
+	{
+		return this.records.values().size();
+	}
+
+	@Override
+	public Iterator<IAEFluidStack> iterator()
+	{
+		return new MeaningfulFluidIterator<IAEFluidStack>( this.records.values().iterator() );
+	}
+
+	@Override
+	public void resetStatus()
+	{
+		for( IAEFluidStack i : this )
+		{
+			i.reset();
+		}
+	}
+
+	private IAEFluidStack getFluidRecord( IAEFluidStack fluid )
+	{
+		final IAEFluidStack fluidrecord = this.records.get( fluid );
+
+		return fluidrecord;
+	}
+
+	private IAEFluidStack putFluidRecord( IAEFluidStack fluid )
+	{
+		return this.records.put( fluid, fluid );
+	}
+
+	private boolean isStackNull( IAEFluidStack stack )
```

Comment:
Already considered it. 

But then they have a point about their description.
It clearly indicates this method will not mutate the current state similar to `final` for variables.
Also only limited to `private`